data/test/macros/wml_unit_test_macros.cfg

```diff
@@ -153,3 +153,94 @@ bob#endarg
 
     {CLEAR_VARIABLE temp}
 #enddef
+
+#define CHECK_STRIKES WEAPONS STRIKES RESULT
+
+#arg ID1
+alice#endarg
+
+#arg ID2
+bob#endarg
+
+#arg COMPARE
+numerical_equals#endarg
+
+    [event]
+        name = start
+
+        [modify_unit]
+            [filter]
+                id={ID1},{ID2}
+            [/filter]
+            [effect]
+                apply_to=attack
+                [set_specials]
+                    mode=append
+                    [damage]
+                        value=1
+                    [/damage]
+                    [chance_to_hit]
+                        value=100
+                    [/chance_to_hit]
+                [/set_specials]
+            [/effect]
+        [/modify_unit]
+
+        {VARIABLE {ID1}_{ID2}_count 0}
+
+        [for]
+            variable=i
+            start=0
+            step=1
+            end="$({WEAPONS}-1)"
+            [do]
+                [test_do_attack_by_id]
+                    attacker={ID1}
+                    defender={ID2}
+                    weapon=$i
+                    resupply_attacks_left=5
+                [/test_do_attack_by_id]
+            [/do]
+        [/for]
+    [/event]
+
+    [event]
+        name=pre attack
```

Comment:
I tried backporting these tests to 1.16, because I wanted to check how much of the weird behavior that I was keeping "because that's what 1.16 did, no point changing it for 1.18 and then changing it again for 1.20" actually matches 1.16's behavior.

`[event]name=pre attack` was only added in 1.17.7, and on 1.16 most of the new tests fail without obvious explanation, because this code doesn't get called, and so the second weapon test sees the count of all strikes including the first weapon's strikes.

Putting the `{VARIABLE {ID1}_{ID2}_strikes 0}` into the `[for][do]` loop makes all 34 tests pass.

Reply:
Personally, I'm not especially concerned with backporting things other than important bug fixes to 1.16, given it'll be oldstable fairly soon.

Reply:
Marking resolved as I've decided that I don't want a change, the main information is that these tests document what 1.16 did.